lib/net/http.rb

```diff
@@ -1615,7 +1615,7 @@ def connect
                                       write_timeout: @write_timeout,
                                       continue_timeout: @continue_timeout,
                                       debug_output: @debug_output)
-          buf = "CONNECT #{conn_address}:#{@port} HTTP/#{HTTPVersion}\r\n"
+          buf = +"CONNECT #{conn_address}:#{@port} HTTP/#{HTTPVersion}\r\n"
           buf << "Host: #{@address}:#{@port}\r\n"
```

Comment:
This should save another string allocation by not storing the `Host` line as an intermediary string, just to add it to `buf`.

```suggestion
          buf = +"CONNECT #{conn_address}:#{@port} HTTP/#{HTTPVersion}\r\n" \
            "Host: #{@address}:#{@port}\r\n"
```